spec/lib/pre_assembly/file_manifest_spec.rb

```diff
@@ -1,9 +1,36 @@
 # frozen_string_literal: true
 
 RSpec.describe PreAssembly::FileManifest do
-  let(:staging_location) { Rails.root.join('spec/fixtures/multimedia') }
+  describe '#load_manifest' do
+    let(:staging_location) { Rails.root.join('spec/fixtures/media_missing') }
+
+    context 'when no rows' do
+      let(:csv_filename) { "#{staging_location}/file_manifest_no_rows.csv" }
+
+      it 'throws an exception' do
+        expect { described_class.new(staging_location:, csv_filename:) }.to raise_error(RuntimeError, 'no rows in file_manifest or missing header')
+      end
+    end
+
+    context 'when invalid rows' do
+      let(:csv_filename) { "#{staging_location}/file_manifest_invalid.csv" }
+
+      it 'throws an exception' do
+        expect { described_class.new(staging_location:, csv_filename:) }.to raise_error(RuntimeError, 'file_manifest has preserve and shelve both being set to no for a single file')
+      end
+    end
+
+    context 'when missing columns' do
+      let(:csv_filename) { "#{staging_location}/file_manifest_missing_columns.csv" }
+
+      it 'throws an exception' do
+        expect { described_class.new(staging_location:, csv_filename:) }.to raise_error(RuntimeError, 'file_manifest missing required columns')
+      end
+    end
+  end
 
```

Comment:
new validation when we load the file manifest: checks for no rows, invalid rows, or missing columns